lib/siwapp/search/search_query.ex

```diff
@@ -3,14 +3,165 @@ defmodule Siwapp.Search.SearchQuery do
   Search Queries
   """
   import Ecto.Query
+  alias Siwapp.Invoices.InvoiceQuery
+  alias Siwapp.{Query, Repo}
+  alias Siwapp.RecurringInvoices.RecurringInvoiceQuery
 
   @doc """
-  Get invoices, customers or recurring_invoices by comparing value with name, email or id fields
+  For each key, one different query
   """
-  def name_email_or_id(query, value) do
+  def filter_by(query, key, value) do
+    case key do
+      "search_input" ->
+        name_email_or_id(query, value)
+
+      "name" ->
+        key = convert_to_atom(key)
+
+        Query.search_in_string(query, key, value)
+
+      "number" ->
+        where(query, [q], q.number == type(^value, :integer))
+
+      "series" ->
+        query
+        |> join(:inner, [q], s in Siwapp.Commons.Series, on: q.series_id == s.id)
+        |> where([q, s], ilike(s.name, ^"%#{value}%"))
+
+      date
+      when date in [
+             "issue_from_date",
+             "issue_to_date",
+             "starting_from_date",
+             "starting_to_date",
+             "finishing_from_date",
+             "finishing_to_date"
+           ] ->
+        value = Date.from_iso8601!(value)
+        type_of_date(query, key, value)
+
+      "status" ->
+        type_of_status(query, value)
+
+      "key" ->
+        where(query, [q], not is_nil(q.meta_attributes[^value]))
+
+      "value" ->
+        keys = get_keys_associated_to_value(query, value)
+
```

Comment:
Estas dos juntas parece que van a hacer muchas queries, son campos que parece tener sentido hacer una query común distinguiendo si cada uno de los campos es vacío o no.  Esta sintaxis es muy nítida: [https://fullstackphoenix.com/tutorials/add-jsonb-field-in-phoenix-and-ecto](url)

Reply:
vale le echo un vistazo

Reply:
Es necesario finalmente hacer dos queries por separado